hibernate-core/src/main/java/org/hibernate/engine/spi/EntityHolder.java

```diff
@@ -0,0 +1,17 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or http://www.gnu.org/licenses/lgpl-2.1.html
+ */
+package org.hibernate.engine.spi;
+
+public interface EntityHolder {
```

Comment:
Is the interface actually useful? I see it's exposed as SPI but if we could avoid that and have a purely internal holder w/o interfaces that might be safer.

Reply:
It is useful and will be extensively used in the next commits. I just wanted to start working on this in baby steps and ensuring that I don't cause any bugs with the changes I do.

Reply:
Javadoc please